crates/bevy_ecs/src/system/commands/parallel_scope.rs

```diff
@@ -0,0 +1,99 @@
+use std::cell::Cell;
+
+use thread_local::ThreadLocal;
+
+use crate::{
+    entity::Entities,
+    prelude::World,
+    system::{SystemParam, SystemParamFetch, SystemParamState},
+};
+
+use super::{CommandQueue, Commands};
+
+#[doc(hidden)]
+#[derive(Default)]
+/// The internal [`SystemParamState`] of the [`ParallelCommands`] type
+pub struct ParallelCommandsState {
+    thread_local_storage: ThreadLocal<Cell<CommandQueue>>,
```

Comment:
I wonder if we should generalize this and add it to `bevy_utils`. This is seemingly a very useful abstraction that showed up in #4899 too.